stderr.go

```diff
@@ -0,0 +1,40 @@
+package errors
+
+import "os"
+
+// IsNotExist checks for os.IsNotExist on provided error,
+// whether it is an *Errors, *Error, or error
+func IsNotExist(err error) bool {
+	return isStdErr(os.IsNotExist, err)
+}
+
+// IsExist checks for os.IsExist on provided error,
+// whether it is an *Errors, *Error, or error
+func IsExist(err error) bool {
+	return isStdErr(os.IsExist, err)
+}
+
+// IsPermission checks for os.IsPermission on provided error,
+// whether it is an *Errors, *Error, or error
+func IsPermission(err error) bool {
+	return isStdErr(os.IsPermission, err)
+}
+
+// isStdErr try-casts err for *Error or *Errors,
+// and checks the underlying error(s) against provided stderrr.
+// If error is not of type *Error or *Errors, isStdErr simply calls stderr(err)
+func isStdErr(stderr func(error) bool, err error) bool {
```

Comment:
coudln't this be exported? I think stderr is a confusing name. could you rename it to something like IsFunc?